lib/guardian_phoenix/socket.ex

```diff
@@ -173,10 +173,9 @@ defmodule Guardian.Phoenix.Socket do
 
   def authenticate(socket, impl, token, claims_to_check, opts) do
     with {:ok, resource, claims} <-
-           Guardian.resource_from_token(impl, token, claims_to_check, opts),
-         key <- Keyword.get(opts, :key, Guardian.Plug.default_key()) do
+           Guardian.resource_from_token(impl, token, claims_to_check, opts) do
+      key = Keyword.get(opts, :key, Guardian.Plug.default_key())
```

Comment:
Why'd you take this out of the with? I'm not sure what this gains?

Reply:
@doomspork I can revert it back, having this in the with is misleading because it will always match so having the statement in the with block causes confusion.

But, let's actually leave the refactoring for another day, I will revert it back.